engine/engine_test.go

```diff
@@ -0,0 +1,76 @@
+package engine
+
+import (
+	"testing"
+
+	"github.com/thrasher-corp/gocryptotrader/config"
+)
+
+func Test_loadConfigWithSettings(t *testing.T) {
+	empty := ""
+	somePath := "somePath"
+	tests := []struct {
+		name     string
+		flags    []string
+		settings *Settings
+		want     *string
+		wantErr  bool
+	}{
+		{
+			name:     "empty",
+			settings: &Settings{},
+			wantErr:  true,
+		},
+		{
+			name: "invalid file",
+			settings: &Settings{
+				ConfigFile: "nonExistent.json",
+			},
+			wantErr: true,
+		},
+		{
+			name: "test file",
+			settings: &Settings{
+				ConfigFile:   config.TestFile,
+				EnableDryRun: true,
+			},
+			want:    &empty,
+			wantErr: false,
+		},
+		{
+			name:  "data dir in settings overrides config data dir",
+			flags: []string{"datadir"},
+			settings: &Settings{
+				ConfigFile:   config.TestFile,
+				DataDir:      somePath,
+				EnableDryRun: true,
+			},
+			want:    &somePath,
+			wantErr: false,
+		},
+	}
+	config.TestBypass = true
+	for _, tt := range tests {
+		tt := tt
```

Comment:
this only helps when t.Parallel() is used so can we add t.Parallel() here pls

Reply:
It's the other way round :) if Parallel() is used, then this definitely needs to be assigned to a local variable, otherwise it also helps to:
a) prevent other potentional issues even without Parallel
b) avoid linter errors

Anyway, these are really small tests so the gain from Parallel is almost non-existant.
Added anyway.

Reply:
Or apparently not...
The global variable config.Cfg is always used when reading files (not sure why or if we could change this), it's not safe to do this in parallel.

Reply:
SORRY bout that! i tested it for the comment above and when i came across another tt := tt i blindly assumed it was a similar context! but yeah t.parallel wont hurt anyways i guess so dont need to revert back or anything